x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2857 @@
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+    <tbody align="center">
+      <tr>
+        <td style="border-right:3px solid black">Keep user customizations when whole string conflicts</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Completely different string</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>Completely different string</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user-added prefixes and accept other modifications</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>[Security Solution] Prebuilt rule name</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>[Security Solution] Updated prebuilt rule name</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user-added suffixes and accept other modifications</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Prebuilt rule name - (Custom version 3)</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name</code></td>
+        <td><code>Updated prebuilt rule name - (Custom version 3)</code></td>
+      </tr>
+      <tr>
+        <td style="border-right:3px solid black">Keep user customizations on conflicts, but add updates in sections with no conflicts</td>
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Customized prebuilt rule name</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name. Much better.</code></td>
+        <td><code>Customized prebuilt rule name. Much better.</code></td>
+      </tr>
+    </tbody>
+  </table>
+
+
+##### Number fields
+
+Number fields should be treated as a whole unit, i.e. not breakable by digits. Therefore, there is only one possibility of conflicts, the scenario (A B C). In that case, **keep the current version** with the user customization.
+
+Examples: `risk_score`, `max_signals`
+
+  <table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+    <tbody align="center">
+      <tr>
+        <td style="border-right:3px solid black">Keep customizations if conflict</td>
+        <td><code>50</code></td>
+        <td><code>65</code></td>
+        <td style="border-right:3px solid black"><code>20</code></td>
+        <td><code>65</code></td>
+      </tr>
+    </tbody>
+  </table>
+
+##### Array fields
+
+###### Array of strings fields
```

Comment:
I think this is too much nesting which makes it hard to navigate the RFC. Look at the TOC preview:

<img width="636" alt="Screenshot 2024-03-18 at 15 02 02" src="https://github.com/elastic/kibana/assets/7359339/76effd93-59d3-47b6-89fa-268f0c2db263">

The following structure would be much simpler and easier to follow:

- Upgrading prebuilt rules
  - Changes to `upgrade/_review` endpoint
  - Changes to `upgrade/_perform` endpoint
  - Changes to diff algorithms
    - Strings
    - Numbers
    - Arrays of strings
    - Arrays of objects
    - MITRE ATT&CK mappings

Reply:
Simplified